common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java

```diff
@@ -0,0 +1,132 @@
+                if (water >= 1) {
+                    float per = (float) water / 16000;
+                    if (per > 1.0F) {
+                        per = 1.0F;
+                    }
+                    int lvl = (int) (per * (float) 54);
+                    if (lvl < 0) {
+                        return;
+                    }
+                    int y = (topPos + 25 + 54) - lvl;
+                    drawTexture(stack, gui, leftPos + 83, y, imageWidth + 28, 54 - lvl, 10, lvl);
+
+                }
+                FluidStack[] outputs = t.getOutputs();
+                int steam = outputs[0].getAmount();
+                if (steam >= 1) {
+                    float per = (float) steam / 16000;
+                    if (per > 1.0F) {
+                        per = 1.0F;
+                    }
+                    int lvl = (int) (per * (float) 54);
+                    if (lvl < 0) {
+                        return;
+                    }
+                    int y = (topPos + 25 + 54) - lvl;
+                    drawTexture(stack, gui, leftPos + 70, y, imageWidth + 18, 54 - lvl, 10, lvl);
+                }
+            });
+            if (container.getTile() instanceof TileEntityLavaBoiler){
+                TileEntityLavaBoiler tile = (TileEntityLavaBoiler)container.getTile();
+                int heat = tile.getHeat();
+                if (heat >= 1) {
+                    float per = (float) heat / tile.getMaxHeat();
+                    if (per > 1.0F) {
+                        per = 1.0F;
+                    }
+                    int lvl = (int) (per * (float) 54);
+                    if (lvl < 0) {
+                        return;
+                    }
+                    int y = (topPos + 25 + 54) - lvl;
+                    drawTexture(stack, gui, leftPos + 96, y, imageWidth + 38, 54 - lvl, 10, lvl);
+                }
+                int fuel = tile.getFuel();
+                if (fuel > 0) {
+                    float per = (float) fuel / tile.getMaxFuel();
+                    if (per > 1.0F) {
+                        per = 1.0F;
+                    }
+                    int lvl = (int) (per * (float) 18);
+                    if (lvl < 0) {
+                        return;
+                    }
+                    int y = (topPos + 42 + 18) - lvl;
+                    drawTexture(stack, gui, leftPos + 115, y, imageWidth, 18 - lvl, 18, lvl);
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int mouseButton) {
+        if ((!AntimatterAPI.isModLoaded(Ref.MOD_JEI) && !AntimatterAPI.isModLoaded(Ref.MOD_REI)) || !container.getTile().has(MachineFlag.RECIPE))
+            return super.mouseClicked(mouseX, mouseY, mouseButton);
+        if (isInGui((imageWidth / 2) - 10, 24, 20, 18, mouseX, mouseY)) {
+            return false;
+        }
+        if (isInGui(115, 43, 18, 18, mouseX, mouseY)) {
```

Comment:
the gui system does this automatically, it selects the appropriate widget from the gui position

Reply:
Actually I don't think either `ScreenLavaBoiler` or `ScreenCoalBoiler` are actually used any more and should probably be removed.

Reply:
I've removed ScreenLavaBoiler as it wasn't doing anything.

I'll remove the ScreenCoalBoiler class in another PR to keep the scope of this one limited.